test/fixture/infinity-stack-trace.js

```diff
@@ -0,0 +1,14 @@
+import test from '../../';
+import {AssertionError} from '../../lib/assert';
+
+test(t => {
+	const limitBefore = Error.stackTraceLimit;
+	Error.stackTraceLimit = 1;
```

Comment:
Since this test is run in its own process you could set the limit at the top level, before the `test()` call.